### PR TITLE
Fix diagnostic se_type bug

### LINKED ISSUE #310: iv_robust, diagnostics, and factor variables in the first stage
When combining a first-stage factor with the `diagnostics=TRUE` option, `iv_robust()` will fail to select a default `se_type`, unless `weights` is also specified. The default SEs work fine, they just must be specified rather than being default.

Edit: and I checked, it's not because I'm overidentified here, and this is estimatr 0.18.0.

```
library(tidyverse)
library(estimatr)
o <- 1000
tb <- tibble(c = sample(1:(o/10),o,replace=T),
             w = rnorm(o),
             groups = rep(c('A','B','C','D'),o/4),
             gamma = 2,
             beta = 3,
             z = rnorm(o),
             nu = rnorm(o),
             eps = rnorm(o)) %>%
  group_by(c) %>%
  mutate(etac = first(rnorm(o)),
         hc = first(z)) %>%
  ungroup() %>%
  mutate(x = z*gamma + hc*(etac+ w +nu)) %>%
  mutate(y = x*beta +  hc*(etac+w +eps))

#Works fine
ivclus <- iv_robust(y~x|z,data=tb,clusters = c)
ivclus <- iv_robust(y~x|z,data=tb,diagnostics=TRUE,clusters = c)
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,clusters = c)
ivclus <- iv_robust(y~x|z+factor(groups),data=tb)

#If I include a factor AND diagnostics:
#Error in if (se_type == "classical") { : argument is of length zero
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,diagnostics=TRUE)
#Error in if (se_type == "classical") { : argument is of length zero
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,diagnostics=TRUE,clusters = c)

#default errors work fine, they just must be specified
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,diagnostics=TRUE,se_type='HC2')
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,diagnostics=TRUE,clusters = c,se_type='CR2')

#or, oddly, they don't have to be specified if I add weights?
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,diagnostics=TRUE,weights = rep(1,nrow(tb)))
ivclus <- iv_robust(y~x|z+factor(groups),data=tb,diagnostics=TRUE,clusters = c,weights = rep(1,nrow(tb)))
```

### PR DESCRIPTION
**Pull request by lukesonnet**
Fixes #310 by explicitly overloading `se_type` before diagnostics are run.

### CODE REVIEW

File: R/estimatr_iv_robust.R [hunk 1]
```diff
@@ -246,6 +246,8 @@ iv_robust <- function(formula,
     formula = formula
   )
 
+  se_type <- return_list[["se_type"]]
```
Comment (nfultz) [inline R/estimatr_iv_robust.R]: Anything else need to be set while you are in here? Or is se_type the exception bc it defaults to NULL?

Comment (lukesonnet) [reply R/estimatr_iv_robust.R]: It's the exception because of NULL!

**Comment by graemeblair** [comment]
What's the status of this @lukesonnet?

**Comment by lukesonnet** [comment]
This should be good and ready to go? I'll try and resolve the Travis issue and then I'll ping you for review, @graemeblair.

**Pull request merged**